libzeth/snarks/default/default_api_handler.hpp

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2015-2020 Clearmatics Technologies Ltd
```

Comment:
Why not: `mv $ZETH/libzeth/snarks/default/* $ZETH/libzeth/snarks/`? (default is already part of the file names)

Reply:
I realized that this was done for all files under `default`, `groth16` and `pghr13`. Why not rename the files from
```
default
|    |_ default_api_handler.hpp
|    |_ default_snark.hpp
|
|_ groth16
|      |_ groth16_api_handler.hpp
|      |_ groth16_snark.hpp
...
```

to 

```
default
|    |_ api_handler.hpp
|    |_ snark.hpp
|
|_ groth16
|      |_ api_handler.hpp
|      |_ snark.hpp
...
```

?

Reply:
Just to be sure, are you suggesting this?
```
  libzeth/
    snarks/
      groth16_snark.*
      groth16_api_handler.*
      pghr13_snark.*
      pghr13_api_handler.*
      default_snark.*
      default_api_handler.*
```
I'm happy to go with that, if it's what you are referring to.

Reply:
Just seen the 2nd comment ... I wanted to avoid having multiple files with the same name (even in different dirs).  It can be confusing, and also doesn't match the pattern of naming files after the main type they correspond to.  (e.g. `groth16_snark` defines the `groth16_snark` type).

Reply:
Hum.. fair. I wouldn't have done it that way, but I am happy to stick to what you proposed.